### PR TITLE
[DNM] Fix issue where setting `TZ` environment variable in TiKV pod doesn't work

### PR DESCRIPTION
**Pull request by weekface**
### What problem does this PR solve? 

fixes: #1148 

### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Helm charts change
 - Has Go code change
 - Has CI related scripts change
 - Has documents change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note
Fix issue where setting `TZ` environment variable in TiKV pod doesn't work
 ```

### CODE REVIEW

**Comment by weekface** [comment]
https://github.com/tikv/tikv/pull/3575 TiDB push down the local timezone to TiKV.

File: pkg/manager/member/tikv_member_manager.go [hunk 1]
```diff
@@ -410,6 +411,10 @@ func getNewTiKVSetForTidbCluster(tc *v1alpha1.TidbCluster) (*apps.StatefulSet, e
 	if tc.Spec.TiKV.HostNetwork {
 		dnsPolicy = corev1.DNSClusterFirstWithHostNet
 	}
+	tz := tc.Spec.Timezone
+	if tz != "UTC" {
```
Comment (aylei) [inline pkg/manager/member/tikv_member_manager.go]:
```suggestion
	if tz != "UTC" && tz != "" {
```
Shall we check for empty string?

File: pkg/manager/member/tikv_member_manager_test.go [hunk 2]
```diff
@@ -140,8 +144,39 @@ func TestTiKVMemberManagerSyncCreate(t *testing.T) {
 			err:                          false,
 			tikvPeerSvcCreated:           true,
 			setCreated:                   true,
-			pdStores:                     &pdapi.StoresInfo{Count: 0, Stores: []*pdapi.StoreInfo{}},
-			tombstoneStores:              &pdapi.StoresInfo{Count: 0, Stores: []*pdapi.StoreInfo{}},
+			expectStatefulSetFn: func(g *GomegaWithT, set *apps.StatefulSet, err error) {
```
Comment (aylei) [inline pkg/manager/member/tikv_member_manager_test.go]: maybe a new case for empty tz

**Comment by weekface** [comment]
- set it to `UTC` if `.spec.Timezone` is empty or `UTC`
- set it to `/usr/share/zoneinfo/<tz>` if `.spec.Timezone` is not `UTC`

@tennix @aylei PTAL again.

File: pkg/manager/member/tikv_member_manager.go [hunk 3]
```diff
@@ -410,6 +411,13 @@ func getNewTiKVSetForTidbCluster(tc *v1alpha1.TidbCluster) (*apps.StatefulSet, e
 	if tc.Spec.TiKV.HostNetwork {
 		dnsPolicy = corev1.DNSClusterFirstWithHostNet
 	}
+	tz := tc.Spec.Timezone
+	if tz == "" {
```
Comment (tennix) [inline pkg/manager/member/tikv_member_manager.go]: Should we apply this logic to other components?

**Comment by weekface** [comment]
Closing, the root problem is https://github.com/tikv/tikv/pull/3575, we will set `TZDIR=/usr/share/zoneinfo` to solve this problem.

**Pull request closed without merging**